internal/command/scale/count_machines.go

```diff
@@ -255,9 +255,9 @@ func computeActions(machines []*fly.Machine, expectedGroupCounts groupCounts, re
 	machineGroups := lo.GroupBy(machines, func(m *fly.Machine) string {
 		return m.ProcessGroup()
 	})
-	expectedCounts := lo.MapValues(expectedGroupCounts, func(c groupCount, _ string) int {
+	expectedCounts := lo.MapValues(expectedGroupCounts, func(c groupCount, group string) int {
 		if c.relative != 0 {
-			return len(machines) + c.relative
+			return len(machineGroups[group]) + c.relative
```

Comment:
does it need a edge case check for non-negative absolute numbers ?

Reply:
not sure what you mean here, can you give an example?

Reply:
Sorry, I wasn't clear. I mean if `c.relative` is negative and larger than `len(machineGroups[group]` the final absolute count for the group will be negative.

I was thinking on wrapping it like: `lo.Max([]int{0, len(machineGroups[group]) - c.relative})` so negative counts aren't possible.

Reply:
Oh, I just tried to run `fly scale count app=-1` (on main branch) and it hung. flyctl process consumes 100%, likely entered a infinite loop.

Reply:
just a suggestion:

```diff
diff --git a/internal/command/scale/count_machines.go b/internal/command/scale/count_machines.go
index 894d22fe..37260dbb 100644
--- a/internal/command/scale/count_machines.go
+++ b/internal/command/scale/count_machines.go
@@ -362,7 +362,7 @@ func convergeGroupCounts(expectedTotal int, current map[string]int, regions []st
                }
        }

-       diff := expectedTotal
+       diff := lo.Max([]int{0, expectedTotal})
        for _, region := range regions {
                diff -= (current[region] + diffs[region])
        }

```

Reply:
ahh ok, it looks like an existing bug, but since we're touching this code it does make sense to fix - a2130157d8a70cdb3c4bcfee0d25d35da6dc4f71